internal/persistence/sql/relationtuples.go

```diff
@@ -138,16 +250,34 @@ func (p *Persister) GetRelationTuples(ctx context.Context, query *relationtuple.
 		wheres = append(wheres, whereStmts{stmt: "object = ?", arg: query.Object})
 	}
 	if query.Subject != nil {
-		wheres = append(wheres, whereStmts{stmt: "subject = ?", arg: query.Subject.String()})
-	}
+		switch s := query.Subject.(type) {
+		case *relationtuple.SubjectID:
+			wheres = append(wheres, whereStmts{stmt: "subject_id = ?", arg: s.ID})
+		case *relationtuple.SubjectSet:
+			n, err := nm.GetNamespaceByName(ctx, s.Namespace)
+			if err != nil {
+				return nil, "", err
+			}
 
-	n, err := p.namespaces.GetNamespace(ctx, query.Namespace)
-	if err != nil {
-		return nil, "", err
+			wheres = append(
+				wheres,
+				whereStmts{stmt: "subject_set_namespace_id = ?", arg: n.ID},
+				whereStmts{stmt: "subject_set_object = ?", arg: s.Object},
+				whereStmts{stmt: "subject_set_relation = ?", arg: s.Relation},
+			)
+		}
+	}
+	if query.Namespace != "" {
+		n, err := nm.GetNamespaceByName(ctx, query.Namespace)
+		if err != nil {
+			return nil, "", err
+		}
+		wheres = append(wheres, whereStmts{stmt: "namespace_id = ?", arg: n.ID})
 	}
 
-	sqlQuery := p.connection(context.WithValue(ctx, namespaceContextKey, n)).
-		Order("object, relation, subject, commit_time").
+	sqlQuery := p.Connection(ctx).
+		Where("nid = ?", p.NetworkID(ctx)).
+		Order("nid, namespace_id, object, relation, subject_id, subject_set_namespace_id, subject_set_object, subject_set_relation").
```

Comment:
These are quite a lot of order operations. Does that really make sense? These would all need to be indexed.

Also, ordering by `nid` doesn't seem logical to me as `nid` is always the same value for every row

Reply:
To get reliable pagination we have to sort things. The `nid` can probably be removed, but I think the idea was to use index prefixes to minimize the amount of (yet missing) indexes. Will add them though and check whether the `nid` has to be included or not.

Reply:
That makes sense, I think we can remove the nid then from here